src/Instances/Attribute.lua

```diff
@@ -0,0 +1,53 @@
+--!strict
+
+--[[
+	A special key for property tables, which allows users to apply custom
+    attributes to instances
+]]
+
+local Package = script.Parent.Parent
+local PubTypes = require(Package.PubTypes)
+local logError = require(Package.Logging.logError)
+local xtypeof = require(Package.Utility.xtypeof)
+local Observer = require(Package.State.Observer)
+
+local function setAttribute(instance: Instance, attribute: string, value: any)
+    instance:SetAttribute(attribute, value)
+end
+
+local function bindAttribute(instance: Instance, attribute: string, value: any, cleanupTasks: {PubTypes.Task})
+    if xtypeof(value) == "State" then
+        local didDefer = false
+        local function update()
+            if not didDefer then
+				didDefer = true
+				task.defer(function()
+					didDefer = false
+					setAttribute(instance, attribute, value:get(false))
+				end)
+			end
+        end
+
+		setAttribute(instance, attribute, value:get(false))
+		table.insert(cleanupTasks, Observer(value :: any):onChange(update))
+    else
+        setAttribute(instance, attribute, value)
+    end
+end
+
+local function Attribute(attributeName: string)
+    local AttributeKey = {}
+    AttributeKey.type = "SpecialKey"
+    AttributeKey.kind = "Attribute"
+    AttributeKey.stage = "self"
+
+    function AttributeKey:apply(attributeValue: any, applyTo: Instance, cleanupTasks: {PubTypes.Task})
+        if attributeName == nil then
+            logError("attributeNameNil")
+        end
```

Comment:
This could be checked directly on calling `Attribute` for a clearer error stack trace.

Reply:
Fixed!